src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
```

Comment:
so what are we doing here? imo having a 108 char limit on the path of the socket is fine but I am open to copying the solution from fastd once that is reworked. :)